### PR TITLE
Fixed isssue #522

### PR DESCRIPTION
**Pull request by user-promi**

### CODE REVIEW

File: api/class-mvx-rest-controller.php [hunk 1]
```diff
@@ -2640,6 +2640,62 @@ public function mvx_bulk_todo_pending_product($request) {
                     }
                 }
             }
+        } elseif ($type == "pending_verification") {
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2643]: checking missing

File: api/class-mvx-rest-controller.php [hunk 2]
```diff
@@ -2640,6 +2640,62 @@ public function mvx_bulk_todo_pending_product($request) {
...
+            if ($this->mvx_fetch_pending_verification_data()->data) {
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php]: checking missing

File: api/class-mvx-rest-controller.php [hunk 3]
```diff
@@ -2640,6 +2640,62 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                foreach ($this->mvx_fetch_pending_verification_data()->data as $key => $value) {
+                    if ($data_list[$key]) {
+                        $get_pending_verification_list[] = $value['id'];
+                    }
+                }
+            }
+            if ($select_option_value == "dismiss") {
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php:2651]: checking missing

File: api/class-mvx-rest-controller.php [hunk 4]
```diff
@@ -2640,6 +2640,62 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                if ($get_pending_verification_list) {
+                    foreach ($get_pending_verification_list as $request_vendor_key => $request_vendor_id) {
+                        $user_id = $request_vendor_id;
+                        $verification_settings = get_user_meta($user_id, 'mvx_vendor_verification_settings', true);
+                        if(isset($verification_settings['id_verification']['is_verified']) && $verification_settings['id_verification']['is_verified'] != 'verified') {
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php]: indentation

File: api/class-mvx-rest-controller.php [hunk 5]
```diff
@@ -2640,6 +2640,62 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                            $verification_settings['id_verification']['data'] = '';
+                            delete_user_meta($user_id, 'mvx_vendor_is_verified');
+                        }
+                        if(isset($verification_settings['address_verification']['is_verified']) &&  $verification_settings['address_verification']['is_verified'] != 'verified') {
+                            $verification_settings['address_verification']['is_verified'] = 'rejected';
+                            $verification_settings['address_verification']['data'] = '';
+                            delete_user_meta($user_id, 'mvx_vendor_is_verified');
+                        }
+                        
+                        update_user_meta($user_id, 'mvx_vendor_verification_settings', $verification_settings);
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php]: validate before update

File: api/class-mvx-rest-controller.php [hunk 6]
```diff
@@ -2640,6 +2640,62 @@ public function mvx_bulk_todo_pending_product($request) {
...
+            
+                        if(isset($verification_settings['address_verification']['is_verified']) && $verification_settings['address_verification']['is_verified'] == 'verified' && isset($verification_settings['id_verification']['is_verified']) && $verification_settings['id_verification']['is_verified'] == 'verified'){
+                            update_user_meta($user_id, 'mvx_vendor_is_verified', 'verified');
+            
+                        }else{
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php]: indentation

File: api/class-mvx-rest-controller.php [hunk 7]
```diff
@@ -2640,6 +2640,62 @@ public function mvx_bulk_todo_pending_product($request) {
...
+                        }
+                        if(isset($verification_settings['address_verification']['is_verified']) &&  $verification_settings['address_verification']['is_verified'] != 'verified') {
+                            $verification_settings['address_verification']['is_verified'] = 'rejected';
+                            $verification_settings['address_verification']['data'] = '';
+                            delete_user_meta($user_id, 'mvx_vendor_is_verified');
+                        }
+                        
+                        update_user_meta($user_id, 'mvx_vendor_verification_settings', $verification_settings);
+            
+                        if(isset($verification_settings['address_verification']['is_verified']) && $verification_settings['address_verification']['is_verified'] == 'verified' && isset($verification_settings['id_verification']['is_verified']) && $verification_settings['id_verification']['is_verified'] == 'verified'){
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php]: indentation

**Comment by moumitahalder** [comment]
There are so many issues. not able to comment everywhere. do focus and concentrate while doing any single line of code. 
@user-promi

**Review by moumitahalder — CHANGES_REQUESTED** [review]

File: api/class-mvx-rest-controller.php (hunk 1 above)
Comment (user-promi) [reply api/class-mvx-rest-controller.php:2643]: checking is there in line no - 2458

File: api/class-mvx-rest-controller.php (hunk 3 above)
Comment (user-promi) [reply api/class-mvx-rest-controller.php:2651]: checking is there in line no - 2457

**Pull request closed without merging**